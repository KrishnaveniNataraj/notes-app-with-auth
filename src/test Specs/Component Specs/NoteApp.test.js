import NotesApp from '../../components/NotesApp';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

let wrapper, notes, handleAddNote, handleRemoveNote;
beforeEach(() => {
    notes = [{
        "noteTitle": "note1",
        "noteDescription": "note1 desc",
        "id": 1
    }]
    handleAddNote = jest.fn();
    handleRemoveNote = jest.fn();
    wrapper = shallow(
        <NotesApp
            notes={notes}
            handleAddNote={handleAddNote}
            handleRemoveNote={handleRemoveNote}
        />
    );
});


test('Render NotesApp component with notes correctly', () => {
    expect(wrapper).toMatchSnapshot();
});