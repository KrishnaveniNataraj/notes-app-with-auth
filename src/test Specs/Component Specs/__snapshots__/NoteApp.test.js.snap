// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NotesApp component with notes correctly 1`] = `
<WithStyles(Grid)
  container={true}
  spacing={8}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(NoteTaker)
      handleAddNote={[MockFunction]}
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <NotesContainer
      handleRemoveNote={[MockFunction]}
      notes={
        Array [
          Object {
            "id": 1,
            "noteDescription": "note1 desc",
            "noteTitle": "note1",
          },
        ]
      }
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
